From: Jiri Slaby <jslaby@suse.cz>
Subject: kgr: handle btrfs kthreads
Patch-mainline: never, SLE12 specific
References: fate#313296 bnc#889207

Add one more kgr_task_safe to btrfs kthread loop. This one is not
upstream, so this is only SUSE specific.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 fs/btrfs/async-thread.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/fs/btrfs/async-thread.c
+++ b/fs/btrfs/async-thread.c
@@ -340,6 +340,8 @@ again:
 		spin_lock_irq(&worker->lock);
 		check_idle_worker(worker);
 
+		kgr_task_safe(current);
+
 		if (freezing(current)) {
 			worker->working = 0;
 			spin_unlock_irq(&worker->lock);
